src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -841,6 +841,78 @@ namespace OpenLoco::Vehicles
+            {
+                car.front->objectSpriteType = 0xFF;
+                car.back->objectSpriteType = 0xFF;
+                car.body->objectSpriteType = 0xFF;
+                auto frontBogieOfNext = car.body->nextVehicleComponent();
+                if (frontBogieOfNext == nullptr)
+                {
+                    throw Exception::RuntimeError("connectJacobsBogies frontBogieOfNext was unexpectedly nullptr");
+                }
+
+                CarComponent nextComponent = CarComponent(frontBogieOfNext);
+                auto o = ObjectManager::get<VehicleObject>(car.front->objectId);
+                nextComponent.front->objectSpriteType = o->carComponents[nextComponent.body->bodyIndex].frontBogieSpriteInd;
+                if (nextComponent.body->has38Flags(Flags38::isReversed))
+                {
+                    nextComponent.front->objectSpriteType = o->carComponents[nextComponent.body->bodyIndex].backBogieSpriteInd;
+                }
+                // connect jacob's bogie to the front
+                if (componentsFound >= 1 && previousCarComponent.body->has38Flags(Flags38::jacobsBogieAvailable))
+                {
+                    if (componentsFound < 2)
+                    {
+                        throw Exception::RuntimeError("connectJacobsBogies tried to connect jacob's bogie without secondPreviousCarComponent");
+                    }
+                    auto o2 = ObjectManager::get<VehicleObject>(car.front->objectId);
+                    car.front->objectSpriteType = o2->carComponents[car.front->bodyIndex].frontBogieSpriteInd;
+                    nextComponent.front->objectSpriteType = 0xFF;
+                    secondPreviousCarComponent.back->objectSpriteType = 0xFF;
+                }
+            }
+            for (auto& component : car)
+            {
+                // reset jacob's bogie at the rear of the car
+                // jacobs bogie flag is only set on the first and last CarComponent of the car, it cannot be set on middle one(s)
+                if (component.body->has38Flags(Flags38::jacobsBogieAvailable) && component.body->getSubType() == VehicleEntityType::body_continued)
+                {
+                    if (componentsFound == 0)
+                    {
+                        throw Exception::RuntimeError("connectJacobsBogies reached end of Car without previousCarComponent");
+                    }
+                    component.front->objectSpriteType = 0xFF;
+                    component.back->objectSpriteType = 0xFF;
+                    component.body->objectSpriteType = 0xFF;
+                    auto o3 = ObjectManager::get<VehicleObject>(previousCarComponent.back->objectId);
+                    previousCarComponent.back->objectSpriteType = o3->carComponents[previousCarComponent.back->bodyIndex].backBogieSpriteInd;
+                    if (previousCarComponent.body->has38Flags(Flags38::isReversed))
```

Comment:
where is this coming from i don't see it in the assembly.

Reply:
It was a CS oversight that I forgot to comment